### PR TITLE
Clustering/round robin check ttl

### LINKED ISSUE #1620: Check TTL should work with round-robin if proxy_entity_id set
Check TTL should work when using round-robin subscription(s) and Check proxy_entity_id is set.

## Expected Behavior
As an operator, I should be able to create a Check with a TTL when using a round-robin subscription if I also specify a proxy_entity_id.

## Current Behavior
Check TTL is currently disabled when round-robin is used, see:

https://github.com/sensu/sensu-go/blob/fe473a9ee28b25e106b8004a404f6e3522485ec1/backend/eventd/eventd.go#L208-L212

Requires #1674

### PR DESCRIPTION
**Pull request by echlebek**
## What is this change?

This pull request deprecates the existing monitor and replaces it with a new one. Eventd and Keepalived have been modified to use the new monitor.

The new monitor does not rely on any local state, but instead uses state from etcd. This enables clustered check TTL and keepalives to work properly, including round robin check TTL.

## Why is this change necessary?

Without this change, clustered Sensu will not work properly. It also makes round robin check TTL work.

Closes #1620 

## Does your change need a Changelog entry?

Yes

### CODE REVIEW

File: backend/monitor/monitor.go [hunk 1]
```diff
@@ -23,160 +22,25 @@ type Interface interface {
...
-	stopped   bool
-	failing   bool
-	mu        sync.Mutex
-}
-
-// UpdateHandler provides an event update handler.
-type UpdateHandler interface {
-	HandleUpdate(e *types.Event) error
+// Handler is a handler for monitor events.
+type Handler interface {
```
Comment (mercul3s) [inline backend/monitor/monitor.go]: 🙌 thank you for interfacing and consolidating the interim changes - looks good!

File: backend/eventd/eventd.go [hunk 2]
```diff
@@ -203,22 +198,16 @@ func (e *Eventd) handleMessage(msg interface{}) error {
...
-		// monitor map
-		// only monitor if there is a check TTL and the check is not a
-		// round robin check.
-		e.mu.Lock()
-		mon, ok = e.monitors[entity.ID]
-		if !ok || mon.IsStopped() {
-			timeout := time.Duration(event.Check.Ttl) * time.Second
-			mon = e.monitorFactory(entity, event, timeout, e, e)
-			e.monitors[entity.ID] = mon
+	if event.Check.Ttl > 0 {
```
[the remark this answers is not in the document]
Comment (echlebek) [reply backend/eventd/eventd.go]: Speaking with @portertech, it seems that a lot of the time people will be using proxy entities for round robin. We need to be able to support that use case.

Not exactly sure what the best way is to handle the case for no proxy entity.

[the remark this answers is not in the document]
Comment (mercul3s) [reply backend/eventd/eventd.go]: Ah, I see. I'm not quite sure the best way to handle that either at this point.

[the remark this answers is not in the document]
Comment (palourde) [reply backend/eventd/eventd.go]: I suppose we could look into the entity class to verify if it's an proxy ID? e.g.

```
if event.Check.Ttl > 0 && (!event.Check.RoundRobin || event.Entity.Class == types.EntityProxyClass) {
```

**Comment by echlebek** [comment]
I think I bit off a little more than I was supposed to with this PR. It closes #1102 and also closes #1103.

**Pull request merged**